extensions/google-java/src/main/kotlin/com/connectrpc/extensions/JavaErrorParser.kt

```diff
@@ -31,30 +33,22 @@ internal object JavaErrorParser : ErrorDetailParser {
      * expects a generic type E? and the underlying type is a Google
      * Java Message.
      */
-    @Suppress("UNCHECKED_CAST")
     override fun <E : Any> unpack(any: AnyError, clazz: KClass<E>): E? {
-        val value = any.value.utf8().decodeBase64() ?: any.value
-        val anyProto = com.google.protobuf.Any.newBuilder()
-            .setTypeUrl(if (any.typeUrl.contains('/')) any.typeUrl else "$TYPE_URL_PREFIX${any.typeUrl}")
-            .setValue(com.google.protobuf.ByteString.copyFrom(value.asByteBuffer()))
-            .build()
-        val kClass = clazz.java as Class<Message>
-        val unpacked = anyProto.unpack(kClass)
-        if (unpacked?.javaClass?.isAssignableFrom(clazz.java) == true) {
-            return unpacked as E?
+        if (!clazz.isSubclassOf(MessageLite::class)) {
+            throw RuntimeException("class ${clazz.qualifiedName} does not extend MessageLite")
         }
-        return null
+        @Suppress("UNCHECKED_CAST") // we just checked above, so it's safe
+        val instance = Internal.getDefaultInstance(clazz.java as Class<MessageLite>) as MessageLite
+        val unpacked = instance.parserForType.parseFrom(any.value.toByteArray())
```

Comment:
I just realized that one thing we lose with the change to directly deserialize, instead of using `Any.unpack`, is that we're no longer validating the message type indicated in the URL matches the given message class. I'll push a change for that...